sonic_installer/bootloader/grub.py

```diff
@@ -81,6 +82,34 @@ def remove_image(self, image):
         run_command('grub-set-default --boot-directory=' + HOST_PATH + ' 0')
         click.echo('Image removed')
 
+    def verify_binary_image(self, image_path):
+        if not os.path.isfile(image_path):
+            return False
+
+        # Get running platform's ASIC
+        sonic_version_yml = open('/etc/sonic/sonic_version.yml', 'r')
+        asic_type = re.search(r"asic_type: (\S+)", sonic_version_yml.read()).group(1)
+        sonic_version_yml.close()
```

Comment:
Could you reuse common library: [https://github.com/Azure/sonic-buildimage/blob/7c9be314e2f390a5f2c522f74128ac5e1e2b6230/src/sonic-py-common/sonic_py_common/device_info.py#L294](https://github.com/Azure/sonic-buildimage/blob/7c9be314e2f390a5f2c522f74128ac5e1e2b6230/src/sonic-py-common/sonic_py_common/device_info.py#L294)